documentation/CAMARA-ICM-examples.md

```diff
@@ -0,0 +1,358 @@
+    &authorization_details=%5B%7B%22type%22%3A%22org.camaraproject.purpose%22%2C%22purpose%22%3A%22dpv%3AFraudPreventionAndDetection%22%7D%5D
+    &scope=openid%20sim-swap:check%20sim-swap:retrieve-date
+    &client_id=s6BhdRkqt3
+    &state=af0ifjsldkj
+    &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb HTTP/1.1
+Host: server.example.com
+```
+
+#### RFC9101 request object `authorization_details` one purpose
+
+```
+{
+  "iss": "s6BhdRkqt3",
+  "aud": "https://server.example.com",
+  "response_type": "code",
+  "client_id": "s6BhdRkqt3",
+  "redirect_uri": "https://client.example.org/cb",
+  "authorization_details": "[{"type":"org.camaraproject.purpose","purpose":"dpv:FraudPreventionAndDetection"}]"
+  "scope": "openid sim-swap:check sim-swap:retrieve-date",
+  "state": "af0ifjsldkj",
+  "nonce": "n-0S6_WzA2Mj",
+  "max_age": 86400
+}
+```
+
+
+### RFC7662 response with two scopes, one purpose
+
+#### Access Token Variant 1 Response on introspecting an access token with string scope
+
+**Note**: The RS might have to split `purpose` and `technical-scope`.
+
+```
+{
+  "active": true,
+  "client_id": "s6BhdRkqt3",
+  "username": "jdoe",
+  "scope": "dpv:FraudPreventionAndDetection#sim-swap:check dpv:FraudPreventionAndDetection#sim-swap:retrieve-date",
+  "sub": "Z5O3upPC88QrAjx00dis",
+  "aud": "https://protected.example.net/resource",
+  "iss": "https://server.example.com/",
+  "exp": 1419356238,
+  "iat": 1419350238
+}
+```
+
+#### Access Token Variant 2 Response on introspecting an access token with scopes object
+
+```
+{
+  "active": true,
+  "client_id": "s6BhdRkqt3",
+  "username": "jdoe",
+  "scopes": {
+    "sim-swap:check": ["dpv:FraudPreventionAndDetection"],
+    "sim-swap:retrieve-date": ["dpv:FraudPreventionAndDetection"]
+  },
+  "sub": "Z5O3upPC88QrAjx00dis",
+  "aud": "https://protected.example.net/resource",
+  "iss": "https://server.example.com/",
+  "exp": 1419356238,
+  "iat": 1419350238
+}
+```
+
+## Specifying two purpose
+
+---
+**Note**
+
+These following examples are only here now to demonstrate extensibility of the different options in requesting `purpose`.
+Please ignore for now otherwise.
+
+---
+
+### Requesting two purpose, two scopes of the same API
+
+#### Two different purpose encoded in scope
+
+```
+GET /authorize?
+    response_type=code
+    &scope=openid%20dpv%3AFraudPreventionAndDetection%23sim-swap%3Acheck%20dpvvertising%23sim-swap%3Aretrieve-date
+    &client_id=s6BhdRkqt3
+    &state=af0ifjsldkj
+    &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb HTTP/1.1
+Host: server.example.com
+```
+
+#### RFC9101 request object two purpose
+
+```
+{
+  "iss": "s6BhdRkqt3",
+  "aud": "https://server.example.com",
+  "response_type": "code",
+  "client_id": "s6BhdRkqt3",
+  "redirect_uri": "https://client.example.org/cb",
+  "scope": "openid dpv:FraudPreventionAndDetection#sim-swap:check dpv:Advertising#sim-swap:retrieve-date",
+  "state": "af0ifjsldkj",
+  "nonce": "n-0S6_WzA2Mj",
+  "max_age": 86400
```

Comment:
```suggestion
  "max_age": 86400,
  "exp": 1419356238,
  "iat": 1419350238
```
Expiry and issue timestamp is required to prevent reuse

Reply:
The "Commit suggestion" button is grey and the suggestion shown as "outdated" in my UI. So, I am resolving this conversation. Please create a new suggestion if the sub-issue is not resolved.